### DISCUSSION TITLE
Strange training/validation curves for Swin transformer SSL pre-training

### DISCUSSION THREAD

**Discussion by fengling0410** [Project-MONAI/MONAI#4735] [Q&A] [answered]
Hi all, I'm trying to reproduce the pre-trained Swin UNETR and would like to apply it to internal CT scans in my institution. I copy exactly the same code from [this address](https://github.com/Project-MONAI/research-contributions/tree/main/SwinUNETR/Pretrain) except the dataset directory part of `data_utils.py`. The datasets I used are LUNA_16, LIDC, TCIA_colon, and TCIA_COVID19, as stated in the SSL Swin transformer paper. I run the code using single gpu with the command `python main.py --use_checkpoint --batch_size=1 --num_steps=10000 --lrdecay --eval_num=100 --logdir='./runs/test' --lr=4e-4 roi_x=96 --roi_y=96 --roi_z=96 --lr_schedule='warmup_cosine'`. However, the training plot seems quite strange:
The total loss and recon loss doesn't seem to decrease as training goes.
<img width="724" alt="Screen Shot 2022-07-20 at 2 11 48 PM" src="https://user-images.githubusercontent.com/89789702/180053052-ab450f1c-70a5-4a0f-850f-c26b136881b9.png">

The validation image is also strange and it seems like the model is not learning useful features:
<img width="755" alt="Screen Shot 2022-07-20 at 2 11 59 PM" src="https://user-images.githubusercontent.com/89789702/180053289-2993efe7-c49d-4d02-bee5-9843297c0c4e.png">

<img width="758" alt="Screen Shot 2022-07-20 at 2 12 09 PM" src="https://user-images.githubusercontent.com/89789702/180053307-40eeb06c-6cd4-4fd9-8101-addc482e7784.png">

May I ask for some advices regarding this? I'm still new to CT scan segmentation and a little confused about situation. If you spotted anything wrong please let me know. I would appreciate any help from the community :)

**Comment by tangy5** [comment, marked as answer]
Hi Fengling410, I'm not sure with this issue. But when batch size is set to 1, the contrastive learning will cause some problem that lead to unstable overall training. You could try increase the batch size and see whether it will stablize the training. Thanks.

**Reply by fengling0410** [reply]
Hi @tangy5, thanks for your reply. Since I'm using Tesla T4, which only works for batchsize=1, I cannot really increase the batch size. But I made use of 4 GPUs for pre-training following your advice and the Swin Transformer works! I guess the reason is that Swin Transformer needs much larger batch size and training iterations than the common ViT structure.
